### PR TITLE
fix: rename variable to fix register tray item

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
It fix registration of provider tray menu, by properly sending `providerId`.
And, also, fix deleting/removing provider menu item, as `Disposable` which created upon registration is not remove command from tray itself.

As `TrayMenu` expects to receive object with field `providerId` , but we send `providerName`.

### Screenshot/screencast of this PR
Without this fix, any tray provider menu item added in to `temp` sub menu:
Screenshot 2023-03-22 at 15 01 12

With this fix tray menu item adde to proper provider sub menu:

Screenshot 2023-03-22 at 14 59 52

### What issues does this PR fix or reference?
n/a

### How to test this PR?

In your extension use `extensionApi.tray.registerProviderMenuItem` function to register command for your provider,
ensure that your command is on proper sub menu of tray menu. (i.e. no under `temp` menu item)

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
works as expected

do you think you could add a unit test ?

we're now trying to beg for unit tests upon bug fixes or new features

**Comment by gbraad** [comment]
While I agree about unit tests, it might need a lot of additional mock to make this work now. Wonder if it is worth the roi at this time. May we park this for later (tech debt)? I'll let Yevhen decide on this.

**Comment by evidolob** [comment]
@benoitf 

I add test, and also provide fix for deleting provider menu item.

Could you review again?

File: packages/main/src/plugin/tray-menu-registry.spec.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
```
Comment (benoitf) [inline packages/main/src/plugin/tray-menu-registry.spec.ts]:
```suggestion
 * Copyright (C) 2023 Red Hat, Inc.
```

File: packages/main/src/tray-menu.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
```
Comment (benoitf) [inline packages/main/src/tray-menu.spec.ts]:
```suggestion
 * Copyright (C) 2023 Red Hat, Inc.
```

**Review by benoitf — APPROVED** [review]
just small typo on the date

File: packages/main/src/tray-menu.spec.ts (hunk 2 above)
Comment (gbraad) [reply packages/main/src/tray-menu.spec.ts]: :-/

**Comment by gbraad** [comment]
We used to run a check for this on CI... which meant every 1st of Jan you would have failing builds and a big changeset with string replacements for the date ;-)

File: packages/main/src/plugin/tray-menu-registry.spec.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  menuRegistry.registerProvider({
+    internalId: 'internalId',
+    id: 'testId',
+  } as ProviderInfo);
+
+  const menuItem = { id: 'itemId' };
+  const disposable = menuRegistry.registerProviderMenuItem('testId', menuItem);
+  disposable.dispose();
+
+  expect(deleteItem.mock.calls).length(1);
```
Comment (gbraad) [inline packages/main/src/plugin/tray-menu-registry.spec.ts:87]: I am not able to run the tests now, but looks OK.

**Review by gbraad — APPROVED** [review]

**Pull request merged**